scripts/test-crds.sh

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env bash
+
+# ------------------------------------------------------------------------------
+# test-crds.sh
+#
+# This script tests that our legacy v1beta1 versions of our CRDs upgrade cleanly
+# to their v1 versions, and that both the legacy and the v1 versions of the CRDs
+# both function with our controllers.
+#
+# Test Workflow
```

Comment:
(non-blocking but this concerns me)

imo elaborate descriptions "how the code works" at the top of a file like this don't age well. Do you think that we could achieve a similar degree of documentation by means of self-documenting code or local comments?

Reply:
There's a lot of "INFO" output and commented sections which I feel do a good job of explaining the code. This was just meant as a legend so you don't have the scan the whole thing, but I can just as easily pull it and people can read the code for themselves.

Reply:
removed in 0eac9952ed92ff1fb26bec4609bdf183e595fc6b